### PR TITLE
Taylor series using Horner's Rule in C programming

### LINKED ISSUE #1060: Adding Taylor series using Horner Rule in C language.
This is an algorithm of Taylor series using Horner rule, Time complexity is O(n).

### PR DESCRIPTION
**Pull request by Anshuvats**
Fixes #1060 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Adding Taylor series using Horner's Rule in C programming

#### Languages Used:
- C

#### Files Added:
- Taylor_series_Horner_rule.c

### CODE REVIEW

**Comment by somya-kapoor** [comment]
@Anshuvats Please remove the extra file in the pr. (there are 8 files in the pr)

File: Taylor_series_Repo/Taylor_series_Horner_rule.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+//****************************************//
+
+
+#include<stdio.h>
+double e(int x,int n)
+{
+
+    static double s;
+    if(n==0)
```
Comment (somya-kapoor) [inline Taylor_series_Repo/Taylor_series_Horner_rule.c]: Add space around ==

File: Taylor_series_Repo/Taylor_series_Horner_rule.c [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+
+/*Time-Complexity:O(n)*/
+
+
+//****************************************//
+
+
+#include<stdio.h>
+double e(int x,int n)
```
Comment (somya-kapoor) [inline Taylor_series_Repo/Taylor_series_Horner_rule.c]: Add space after comma

File: Taylor_series_Repo/Taylor_series_Horner_rule.c [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+{
+
+    static double s;
+    if(n==0)
+    {
+
```
Comment (somya-kapoor) [inline Taylor_series_Repo/Taylor_series_Horner_rule.c]: Remove line

File: Taylor_series_Repo/Taylor_series_Horner_rule.c [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        return s;
+    }
+    s=1+x*(s/n);
```
Comment (somya-kapoor) [inline Taylor_series_Repo/Taylor_series_Horner_rule.c]: Add space around =,+,*,/

File: Taylor_series_Repo/Taylor_series_Horner_rule.c [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    return e(x,n-1);
```
Comment (somya-kapoor) [inline Taylor_series_Repo/Taylor_series_Horner_rule.c]: Add space after comma

File: Taylor_series_Repo/Taylor_series_Horner_rule.c [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+}
+int main()
+{
+    int x,n;
```
Comment (somya-kapoor) [inline Taylor_series_Repo/Taylor_series_Horner_rule.c]: Add space after comma

File: Taylor_series_Repo/Taylor_series_Horner_rule.c [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+    scanf("%d%d",&x,&n);
```
Comment (somya-kapoor) [inline Taylor_series_Repo/Taylor_series_Horner_rule.c]: Add space after comma

File: Taylor_series_Repo/Taylor_series_Horner_rule.c [hunk 8]
```diff
@@ -40,22 +40,21 @@ f(x0) = a0 + x0(a1 + x0(a2 + x0(a3 + x0(a4 + a5x0))))*/
 double e(int x,int n)
 {
 
-    static double s;
-    if(n==0)
+    static double s; //declare static variable
+    if(n==0)//base case
```
Comment (somya-kapoor) [inline Taylor_series_Repo/Taylor_series_Horner_rule.c]: Add space around ==

File: Taylor_series_Repo/Taylor_series_Horner_rule.c [hunk 9]
```diff
@@ -40,22 +40,21 @@ f(x0) = a0 + x0(a1 + x0(a2 + x0(a3 + x0(a4 + a5x0))))*/
...
-    if(n==0)
+    static double s; //declare static variable
+    if(n==0)//base case
     {
 
         return s;
     }
-    s=1+x*(s/n);
-    return e(x,n-1);
+    s=1+x*(s/n); //General formula comes from Horner rule
```
Comment (somya-kapoor) [inline Taylor_series_Repo/Taylor_series_Horner_rule.c]: Add space around =,+,/,*

File: Taylor_series_Repo/Taylor_series_Horner_rule.c [hunk 10]
```diff
@@ -40,22 +40,21 @@ f(x0) = a0 + x0(a1 + x0(a2 + x0(a3 + x0(a4 + a5x0))))*/
...
+    return e(x,n-1);//Recursive case.
```
Comment (somya-kapoor) [inline Taylor_series_Repo/Taylor_series_Horner_rule.c]: Add space after comma

Comment (somya-kapoor) [reply Taylor_series_Repo/Taylor_series_Horner_rule.c]: After doing the required changes, please squash the commits

File: Taylor_series_Repo/Taylor_series_Horner_rule.c (hunk 3 above)
Comment (Anshuvats) [reply Taylor_series_Repo/Taylor_series_Horner_rule.c]: > @Anshuvats Please remove the extra file in the pr. (there are 8 files in the pr)

@somya-kapoor, I have already squashed this into one commit. Then, please help me, how can I remove this file.

Comment (Anshuvats) [reply Taylor_series_Repo/Taylor_series_Horner_rule.c]: > Remove line
@somya-kapoor  If I remove this line, the base case will delete. then, what is the termination condition of my recursive solution code? If you have some idea about this. please, help me.

Comment (somya-kapoor) [reply Taylor_series_Repo/Taylor_series_Horner_rule.c]: I am talking about line 46 if you removed it nothing will happen to the code.

Comment (somya-kapoor) [reply Taylor_series_Repo/Taylor_series_Horner_rule.c]: Please see https://stackoverflow.com/questions/12481639/remove-files-from-git-commit/12481977
If you still face the issue, we can discuss it on slack.

Comment (Anshuvats) [reply Taylor_series_Repo/Taylor_series_Horner_rule.c]: > Please see https://stackoverflow.com/questions/12481639/remove-files-from-git-commit/12481977
> If you still face the issue, we can discuss it on slack.

 I am not able to squash the commit. please, help me @somya-kapoor. please, reach out me on slack.

**Comment by MastersAbh** [comment]
Hey @Anshuvats , Multiple files have been committed in your PR, please remove them from your commit.
You should only commit files written by you.

**Comment by Anshuvats** [comment]
@MastersAbh I am also upset about this, I have tried a lot of times, but these  8 files are not removing.

**Comment by jainaman224** [comment]
@Anshuvats Please connect with us on slack. We will help you figure it out.

**Comment by MastersAbh** [comment]
Hi,
Can you DM me on slack?
I will be happy to help you out with this.

**Comment by Anshuvats** [comment]
Hi,@MastersAbh! yes, why not.
Thanks a lot. I don't know about this. How can I manage this after many times squashing?
Please Help me.

**Pull request closed without merging**